### DISCUSSION TITLE
hjkl navigation in dialog window like rename

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [JetBrains/ideavim#972] [Q&A] [answered]
Hi all,

I read through the discussion section and haven't found anything about it.

Use case:

- there is a interface I'd like to rename
- move the cursor over it
- hit the "rename combo" defined in `~/.ideaVimrc`
- Rename dialog comes up
  - it has the interface name in an input box and it is selected
  - I want to navigate in this input box using `hjkl` keys

An alternative scenario, which I don't like, would be just type the changes to the interface and using the "Apply changes" function from the Context menu. It doesn't feel comfortable, it is not "the vim way". :)

**Comment by Andras-Csanyi** [comment, marked as answer]
https://x.com/citizenmatt/status/1825219309536526420

**Reply by citizenmatt** [reply]
To expand on this link a little - IdeaVim is an editor extension. That is, it changes the way the editor works. If it’s not an editor, there’s not much IdeaVim can do - what do `h`, `j`, `k` and `l` do unless you’re editing text? It also doesn’t work in arbitrary text fields, because that’s a different control to the main text editor. The text editor is used in some dialogs and tool windows, however, and IdeaVim can work in some of these, such as the Commit tool window. To make these more intuitive, IdeaVim starts in Insert mode, but hitting `<Escape>` will usually close the dialog. You can try alternative shortcuts such as `<C-C>` and (IIRC) `<C-[>` to switch modes. Also check out the `'ideavimsupport'` option in the [docs](https://github.com/JetBrains/ideavim/wiki/set-commands#ideavim-only-options). 

Some of these things might work on Rider’s rename dialog - I’m not sure if that’s a normal text field or an instance of the text editor. Usually, if there’s completion, it’s the text editor. 

However, I don’t think there’s anything wrong with using Vim controls to edit the name of the declaration, and then alt-enter to apply the rename refactoring. The nice thing about IdeaVim is that you get the best of both worlds - IDE and Vim. You can pick and choose the way that works best for you.
